sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -186,8 +186,9 @@ class OrcFileFormat
       if (requestedColIdsOrEmptyFile.isEmpty) {
         Iterator.empty
       } else {
-        val requestedColIds = requestedColIdsOrEmptyFile.get
-        assert(requestedColIds.length == requiredSchema.length,
+        val requestedColIds =
+          requestedColIdsOrEmptyFile.get ++ Array.fill(partitionSchema.length)(-1)
```

Comment:
This semantic change also affects [non-vectorized code path](https://github.com/apache/spark/pull/23387/files#diff-8e7e05590803a656e46183bc94642ab8R229). Can we isolate the scope of this PR?